### DISCUSSION TITLE
Regarding imposition of the boundary conditions (pre_correc function)

### DISCUSSION THREAD

**Discussion by MarcoArtiano** [xcompact3d/Incompact3d#156] [Q&A] [answered]
Hi guys, I'm studying some of the higher order runge kutta methods and some of them are no low-storageable. I'm trying to make the code run using a classic algorithm for the runge kutta for these schemes, but I don't clearly understand how to properly modify the coefficients used in the pre_correc function when a classic runge-kutta algorithm with butcher tableau is employed. Is there any reference or paper who explains the math behind the pre_correc function ? Because I thought the gdt were the c_i *dt in the butcher tableau, but it seems not. 

Also, why am I getting different results in the min max velocity field when I run with ./xcompact3d and mpirun -np 2 ./xcompact3d ?

Thank you so much in advance.

**Comment by slaizet** [comment, marked as answer]
Hi,

The RK3 scheme implemented in Xcompact3d can be found in [Williamson, J. H. (1980). Low-storage runge-kutta schemes. Journal of computational physics, 35(1), 48-56.]. Attached is a small document with more information, this is from [Peyret, R. (2000). Introduction to high-order approximation methods for computational fluid dynamics. In Advanced Turbulent Flow Computations (pp. 1-79). Springer Vienna.] 
[RK.pdf](https://github.com/xcompact3d/Incompact3d/files/10896770/RK.pdf)

In pre_correc, we simply impose the boundary condition on the intermediate velocity field (as this cannot be done on u^{n+1} without loosing the incompressibility). It corresponds to equation 3 in [Laizet, S., & Lamballais, E. (2009). High-order compact schemes for incompressible flows: A simple and efficient method with quasi-spectral accuracy. Journal of Computational Physics, 228(16), 5989-6015.].

Regarding your last question, it is likely due to the random noise in the initial condition (not sure which flow you are talking about).

Best,

Sylvain

**Reply by MarcoArtiano** [reply]
Thanks so much I solved, there was some minor problems on what I've written, it's all perfect now.

**Reply by slaizet** [reply]
Excellent to hear!
Best,
Sylvain
